integration-tests/capacity/transactions.test.ts

```diff
@@ -22,555 +22,604 @@ import {
-                    message: /Custom error: 3/,
-                })
-            });
+  describe("pay_with_capacity", function () {
+    describe("when caller has a Capacity account", async function () {
+      let capacityKeys: KeyringPair;
+      let capacityProvider: u64;
+      let schemaId: u16;
+      const amountStaked = 2n * DOLLARS;
+
+      before(async function () {
+        capacityKeys = createKeys("CapacityKeys");
+        capacityProvider = await createMsaAndProvider(capacityKeys, "CapacityProvider", FUNDS_AMOUNT);
+
+        // Create schemas for testing with Grant Delegation to test pay_with_capacity
+        schemaId = await getOrCreateGraphChangeSchema();
+        assert.notEqual(schemaId, undefined, "setup should populate schemaId");
+      });
+
+      async function assertAddNewKey(addKeyPayload: AddKeyData, newControlKeypair: KeyringPair) {
+        const addKeyPayloadCodec: Codec = ExtrinsicHelper.api.registry.createType("PalletMsaAddKeyData", addKeyPayload);
+        const ownerSig: Sr25519Signature = signPayloadSr25519(capacityKeys, addKeyPayloadCodec);
+        const newSig: Sr25519Signature = signPayloadSr25519(newControlKeypair, addKeyPayloadCodec);
+        const addPublicKeyOp = ExtrinsicHelper.addPublicKeyToMsa(capacityKeys, ownerSig, newSig, addKeyPayload);
+        const [_foo, chainEvents] = await addPublicKeyOp.signAndSend(-1);
+        assertEvent(chainEvents, "system.ExtrinsicSuccess");
+        assertEvent(chainEvents, "msa.PublicKeyAdded");
+      }
+
+      it("fails when a provider makes an eligible extrinsic call using non-funded control key", async function () {
+        // this will first fund 'capacityKeys' before staking
+        await assert.doesNotReject(stakeToProvider(capacityKeys, capacityProvider, amountStaked));
+
+        // As current owner, add a new set of control keys that do not have a balance.
+        let newControlKeypair = createKeys("NewKeyNoBalance");
+        const newPublicKey = newControlKeypair.publicKey;
+        const addKeyPayload: AddKeyData = await generateAddKeyPayload(
+          { msaId: capacityProvider, newPublicKey: newPublicKey }
+        );
+        await assertAddNewKey(addKeyPayload, newControlKeypair);
+
+        // attempt a capacity transaction using the new unfunded key: claimHandle
+        const handle = "test_handle";
+        const expiration = (await getBlockNumber()) + 10;
+        const handle_vec = new Bytes(ExtrinsicHelper.api.registry, handle);
+        const handlePayload = {
+          baseHandle: handle_vec,
+          expiration: expiration,
+        };
+        const claimHandlePayload = ExtrinsicHelper.api.registry.createType("CommonPrimitivesHandlesClaimHandlePayload", handlePayload);
+        const claimHandle = ExtrinsicHelper.claimHandle(newControlKeypair, claimHandlePayload);
+        const [_bar, chainEvents2] = await claimHandle.payWithCapacity();
+        assertEvent(chainEvents2, "system.ExtrinsicFailed");
```

Comment:
Question: Does this also trigger any cost at all?

Aka, is this a DoS vector:
- Create provider
- Add tokenless key to provider
- Get small amount of Capacity
- Flood the chain with failing capacity calls

Reply:
It doesn't even get into the mempool I think but I will double check.  The error is "Invalid Transaction"